runtime/src/main/java/org/corfudb/runtime/view/workflows/WorkflowRequest.java

```diff
@@ -141,6 +141,9 @@ public void invoke() {
                     throw new IllegalStateException("Orchestrator can not be selected");
                 }
 
+            } catch(RuntimeException e) {
```

Comment:
You need to check for the root cause. If the root cause is not an instance of `TImeoutException`, you will want to re-throw.

Reply:
Also, since Corfu's exception handling is pretty screwed up, it might make sense to make exception checking more generic:

```
            } catch (TimeoutException e) {
                log.warn("WorkflowRequest: Error while running {} on attempt {}, cause:", this, x, e);
            } catch (RuntimeException runtimeException) {
                handleException(runtimeException);
            }

    private void handleException(RuntimeException runtimeException) {
        final Throwable cause = ExceptionUtils.getRootCause(runtimeException);
        if (cause instanceof TimeoutException ||
                cause instanceof IllegalStateException ||
                cause instanceof NetworkException) {
            log.warn("WorkflowRequest: Error while running {} with cause:", this, runtimeException);
        } else {
            throw runtimeException;
        }
    }
```

Reply:
1. The root cause was an instance of TimeoutException, coming from bootstrapManagementServer(). 
2. I think its a good idea to add a method listed as above to handle the exceptions gracefully. I will make those changes.